pkg/reconciler/workload/placement/placement_reconcile_scheduling.go

```diff
@@ -51,21 +64,44 @@ func (r *placementSchedulingReconciler) reconcile(ctx context.Context, placement
 	currentScheduled, foundScheduled := placement.Annotations[workloadv1alpha1.InternalSyncTargetPlacementAnnotationKey]
 
 	// 2. pick all valid synctargets in this placements
-	syncTargetClusterName, syncTargets, err := r.getAllValidSyncTargetsForPlacement(clusterName, placement)
-	if err != nil {
-		return reconcileStatusStop, placement, err
+	result := r.getAllValidSyncTargetsForPlacement(ctx, clusterName, placement)
+	if result.err != nil {
+		conditions.MarkFalse(placement,
+			schedulingv1alpha1.PlacementScheduled,
+			schedulingv1alpha1.ScheduleErrorReason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"Failed to schedule: %v",
+			result.err,
+		)
+		return reconcileStatusStop, placement, result.err
 	}
 
 	// no valid synctarget, clean the annotation.
-	if foundScheduled && len(syncTargets) == 0 {
-		expectedAnnotations[workloadv1alpha1.InternalSyncTargetPlacementAnnotationKey] = nil
-		updated, err := r.patchPlacementAnnotation(ctx, clusterName, placement, expectedAnnotations)
-		return reconcileStatusContinue, updated, err
+	if len(result.validSyncTargets) == 0 {
+		var err error
+		if foundScheduled {
+			expectedAnnotations[workloadv1alpha1.InternalSyncTargetPlacementAnnotationKey] = nil
+			placement, err = r.patchPlacementAnnotation(ctx, clusterName, placement, expectedAnnotations)
```

Comment:
this will override other reconciler's changes, won't it? Intentional?

Reply:
and worse: if the patch is a noop (can it be?) it won't trigger a requeue.

Reply:
I am a bit confused, why do we need to explicit requeue after patch? if it is a noop patch, I think we do not expect requeue?

Reply:
The patch func returns the updated object, which is a fresh copy of placement from etcd. Other reconcilers here in the controller might have other changes "queued" up, and those will get lost by `placement = updated` through this line.